### PR TITLE
ci(gcb): ui tweaks to the public logs index.html page

### PR DESCRIPTION
**Pull request by devjgm**
A few small UI tweaks to make the public logs HTML page a bit nicer looking.

Example: https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/6397/2fb9207076ba450bd67750d50acaca6b8f24d4a1/index.html

I'm no UI expert, so this could certainly be improved. Ideas to improve it:

1. Have all pages use an external css file that we store in a single global place (maybe the top of the logs bucket), so we can tweak the UIs of all pages, including old ones, as necessary.
2. We could change the log path to include the exact trigger name so that we can extract that and show users the exact command to locally reproduce a build
3. Any other UI tweaks to make things look nice -- I'm already overstepping my bounds.

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6397)

### CODE REVIEW

**Comment by vnvo2409** [comment]
I've noticed that `Public Access Build Logs` is `http` instead of `https`. The `https` link is working well. Could you change it to `https` as well. I think the line is here
https://github.com/googleapis/google-cloud-cpp/blob/0f25e284d1fc3530108ce604a4cdbee96b28a292/ci/cloudbuild/builds/log-linker.sh#L36

**Comment by devjgm** [comment]
> I've noticed that `Public Access Build Logs` is `http` instead of `https`. The `https` link is working well. Could you change it to `https` as well. I think the line is here
> https://github.com/googleapis/google-cloud-cpp/blob/0f25e284d1fc3530108ce604a4cdbee96b28a292/ci/cloudbuild/builds/log-linker.sh#L36

Fixed. thanks for pointing that out!

File: ci/cloudbuild/notifiers/logs/function/function.cc [hunk 1]
```diff
@@ -17,17 +17,78 @@
...
+// status UI shows trigger names, like `clang-tidy-pr`. To make it easier for
+// customers to correlate these logs w/ the builds that fail in the GitHub UI,
+// we split the logfile component into the separate build parts so we can
+// render links with the same name.
+struct BuildInfo {
+  std::string distro;
+  std::string script;
+  std::string trigger;
+};
+BuildInfo MakeBuildInfo(std::string const& distro_script) {
```
Comment (coryan) [inline ci/cloudbuild/notifiers/logs/function/function.cc:78]: nit: missing blank line.

**Review by coryan — APPROVED** [review]
LGTM, a formatting nit.

I made some notes on an earlier version, ignore them if they show up in the review (I think they shouldn't, but computers are weird...)

File: ci/cloudbuild/notifiers/logs/function/function.cc [hunk 2]
```diff
@@ -17,17 +17,68 @@
...
+// status UI shows trigger names, like `clang-tidy-pr`. To make it easier for
+// customers to correlate these logs w/ the builds that fail in the GitHub UI,
+// we split the logfile component into the separate build parts so we can
+// render links with the same name.
+struct BuildName {
+  std::string distro;
+  std::string script;
+  std::string trigger;
+};
+BuildName MakeBuildName(std::string const& distro_script) {
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/cloudbuild/notifiers/logs/function/function.cc]: Ignore this.

File: ci/cloudbuild/notifiers/logs/function/function.cc [hunk 3]
```diff
@@ -72,47 +123,56 @@ void index_build_logs(CloudEvent event) {  // NOLINT
...
     std::ostringstream os;
     os << "<!DOCTYPE html>\n";
-    os << "<html><meta charset=\"utf-8\"><head>Logs for #" << pr_number << " @ "
-       << commit_sha << "</head>\n";
+    os << "<html><head><meta charset=\"utf-8\"></head>\n";
     os << "<body>\n";
-    os << "<ul>";
+    os << "<h1>Public Build Logs</h1>";
+    write_table(os, v);
+    os << "<ul>\n";
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/cloudbuild/notifiers/logs/function/function.cc]: Ignore this.

File: ci/cloudbuild/notifiers/logs/function/function.cc (hunk 1 above)
Comment (coryan) [reply ci/cloudbuild/notifiers/logs/function/function.cc:78]: Dup.

File: ci/cloudbuild/notifiers/logs/function/function.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/notifiers/logs/function/function.cc]: This function is gone now.

[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/notifiers/logs/function/function.cc]: (saw the "ignore this" after I'd already not ignored it.)

You're right. Sadly, my personal preferences are irrelevant here. This should probably follow the same style as the rest of the repo it's in, so I renamed `index_build_logs` -> `IndexBuildLogs` and updated the README.md accordingly.

**Review by coryan — APPROVED** [review]

**Comment by devjgm** [comment]
`cmake-oldest-deps-pr` failed for an unrelated reason. I'll file an issue about it and merge this change anyway.

* GCB: https://console.cloud.google.com/cloud-build/builds;region=global/8e8ddfbf-bec5-4ae9-8f77-d1f4c7537037;tab=detail?project=cloud-cpp-testing-resources
* Raw: https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/6397/629292ed28bcb099d7c2e0b727777d40c5048afd/fedora-cmake-oldest-deps/log-8e8ddfbf-bec5-4ae9-8f77-d1f4c7537037.txt

**Pull request merged**